func/uk/gov/hmrc/vatapi/resources/ValueAddedTaxReturnsSpec.scala

```diff
@@ -54,6 +55,25 @@ class ValueAddedTaxReturnsSpec extends BaseFunctionalSpec {
         .responseContainsHeader("Receipt-TimeStamp", "2018-02-14T09:32:15Z")
     }
 
+    "allow users to submit VAT returns for non bad_request NRS response" in {
+      given()
+        .stubAudit
+        .userIsFullyAuthorisedForTheNrsDependantResource
+        .nrs().nrsFailureforNonBadRequest(vrn)
+        .des().vatReturns.expectVatReturnSubmissionFor(vrn)
+        .when()
+        .post(s"/$vrn/returns", Some(Json.parse(body())))
+        .withHeaders("Authorization", "Bearer testtoken")
+        .thenAssertThat()
+        .statusIs(201)
+        .bodyHasPath("\\paymentIndicator", "BANK")
+        .bodyHasPath("\\processingDate", "2018-03-01T11:43:43.195Z")
+        .bodyHasPath("\\formBundleNumber", "891713832155")
+        .responseContainsHeader("Receipt-Id", "")
+        .responseContainsHeader("Receipt-Signature", "NOT CURRENTLY IMPLEMENTED")
+        .responseContainsHeader("Receipt-TimeStamp", s"${DateTime.now().toString("yyyy-MM-dd'T'HH:mm:ss'Z'")}")
```

Comment:
unsafe usage of DateTime.now() - since you've rounded to the nearest second the only reason the test is passing is that the time between when the NRSData case class is created (which defaults the timestamp to DateTime.now()) and this assertion is less than one second (spotted by Mark) - might fail on slower environments like a build server